tracer/test/Datadog.Trace.Tests/Configuration/DynamicConfigurationTests.cs

```diff
@@ -104,6 +107,66 @@ public void EnableTracing()
             TracerManager.Instance.Settings.TraceEnabled.Should().BeTrue();
         }
 
+        [Fact]
+        public void SetSamplingRules()
+        {
+            var tracerSettings = new TracerSettings();
+            TracerManager.ReplaceGlobalManager(new ImmutableTracerSettings(tracerSettings), TracerManagerFactory.Instance);
+
+            // sampling rules is null by default
+            TracerManager.Instance.Settings.RemoteSamplingRules.Should().BeNull();
+
+            var traceSampler = TracerManager.Instance.PerTraceSettings.TraceSampler as TraceSampler;
+            traceSampler!.GetRules().Should().ContainSingle().And.AllBeOfType<AgentSamplingRule>();
+
+            const string samplingRulesJson = """
+                                             [{
+                                                "sample_rate": 0.5,
+                                                "provenance": "customer",
+                                                "service": "Service1",
+                                                "resource": "Resource1"
+                                             },
+                                             {
+                                                "sample_rate": 0.1,
+                                                "provenance": "dynamic",
+                                                "service": "Service2",
+                                                "resource": "Resource2"
+                                             }]
+                                             """;
+
+            // set sampling rules "remotely"
+            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig(("tracing_sampling_rules", samplingRulesJson)));
+            TracerManager.Instance.Settings.RemoteSamplingRules.Should().Be(samplingRulesJson);
+
+            traceSampler = TracerManager.Instance.PerTraceSettings.TraceSampler as TraceSampler;
+            var rules = traceSampler!.GetRules();
+            rules.Should().HaveCount(3);
```

Comment:
Just FYI, the `HaveCount()` test is often almost detrimental, because it doesn't show you what you have in the collection, it just tells you that there's the wrong number. A better option _generally_ is something like `rules.Should().BeEquivalentTo([new RemoteCustomSamplingRules(), ..., ..])` etc. Though as you're not checking equivalence of `AgentSamplingRule` anyway, I don't think you can do that here, and you need to avoid the null refs, so 🤷

Reply:
Thanks. I'll see if I checking equivalence of `AgentSamplingRule` works so can I use your suggestion instead of `HaveCount()`.

Reply:
That worked. I removed `HaveCount()` and used `rules.Should().BeEquivalentTo(new ISamplingRule[]{new RemoteCustomSamplingRules(), ..., ...})` instead.